imap_processing/spice/time.py

```diff
@@ -1 +1,72 @@
 """Time conversion functions that rely on SPICE."""
+
+import typing
+from collections.abc import Collection
+from typing import Union
+
+import numpy as np
+import numpy.typing as npt
+import spiceypy as spice
+
+from imap_processing.spice import IMAP_SC_ID
+from imap_processing.spice.kernels import ensure_spice
+
+TICK_DURATION = 2e-5  # 20 microseconds
+
+
+def met_to_j2000ns(
+    met: npt.ArrayLike,
+) -> npt.NDArray[np.int64]:
+    """
+    Convert mission elapsed time (MET) to nanoseconds since J2000.
+
+    Parameters
+    ----------
+    met : array_like
+        Number of seconds since epoch according to the spacecraft clock.
+
+    Returns
+    -------
+    array_like or scalar, int64
+        The mission elapsed time converted to nanoseconds since the J2000 epoch.
+
+    Notes
+    -----
+    There are two options when using SPICE to convert from SCLK time (MET) to
+    J2000. The conversion can be done on SCLK strings as input or using double
+    precision continuous spacecraft clock "ticks". The latter is more accurate
+    as it will correctly convert fractional clock ticks to nanoseconds. Since
+    some IMAP instruments contain clocks with higher precision than 1 SCLK
+    "tick" which is defined to be 20 microseconds, according to the sclk kernel,
+    it is preferable to use the higher accuracy method.
+    """
+    sclk_ticks = np.asarray(met, dtype=float) / TICK_DURATION
+    return np.asarray(_sct2e_wrapper(sclk_ticks) * 1e9, dtype=np.int64)
```

Comment:
Why do we want dtype np.int64 if _sct2e_wrapper returns a float?

Reply:
We are always passing in an array of items too, so do we need the option of a float return and special casing that versus iterating over the array?

Reply:
> Why do we want dtype np.int64 if _sct2e_wrapper returns a float?

This function is specific to returning nanoseconds since the J2000 epoch for use in CDF. The CDF datatype for epoch is int64. Paul Janzen has confirmed that integer nanoseconds has high enough precision.

> We are always passing in an array of items too, so do we need the option of a float return and special casing that versus iterating over the array?

I can certainly add that. I replicated exactly what was in the `met_to_j2000ns` function here but agree that returning a single `int64` instead of an array of length 1. Is that what you meant?

Reply:
> I can certainly add that. I replicated exactly what was in the met_to_j2000ns function here but agree that returning a single int64 instead of an array of length 1 is preferable. Is that what you meant?

I actually meant to always return an array from the underlying wrapped function. You have an `isinstance()` check in there that returns a single float if it is only one value, but here we are always wrapping the input in `np.asarray()` and thus will never hit the single float path from this call location.

Reply:
> I actually meant to always return an array from the underlying wrapped function.

Oh, Ok. I suspect that the underlying wrapper will be used in other time conversion functions, so I would propose that we leave it as is unless there is an argument for always returning an array. I try to follow rather than lead on function API design philosophy. I was thinking that there are numpy functions which return values that look like the input rather than always converting to an array, so I thought it made sense to do that here too.